File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.java
class Solution {
    public List<Integer> getRow(int n) {
        int[] arr = new int[n + 1];
        
        arr[0] = 1;
                
        for(int i = 1; i <= n; i++)
        {
            for(int j = i; j > 0; j--)
            {
                arr[j] += arr[j - 1];
            }
        }
            
        List<Integer> answer = new ArrayList();
        for(int i = 0; i <= n; i++)
        {
            if(arr[i] == 0)
                break;
            
            answer.add(arr[i]);            
        }
        
        return answer;
    }
}

File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.py
class Solution(object):
    def getRow(self, rowIndex):                        
            if rowIndex == 0:
                return [1]
            
            elif rowIndex == 1:
                return [1, 1]
            
            answer = []
            answer.append([1])
            answer.append([1, 1])            
                                    
            for i in range(2, rowIndex + 1):
                subList = [1]                
                prev = answer[-1]
                
                for j in range(1, i):
                    n = prev[j] + prev[j -1]
                    subList.append(n)
                    
                subList.append(1)
                answer.append(subList)
            
            return answer[-1]
            
        
        
        
        